src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java

```diff
@@ -206,35 +207,47 @@ public void sendContestedGeneralApplicationOutcomeEmail(CallbackRequest callback
             recipientEmail = (String) courtDetails.get(COURT_DETAILS_EMAIL_KEY);
         }
 
-        notificationRequest = createNotificationRequest(callbackRequest);
-        notificationRequest.setNotificationEmail(recipientEmail);
+        applicantNotificationRequest = createNotificationRequestForAppSolicitor(callbackRequest);
+        applicantNotificationRequest.setNotificationEmail(recipientEmail);
 
         URI uri = buildUri(notificationServiceConfiguration.getContestedGeneralApplicationOutcome());
-        sendNotificationEmail(notificationRequest, uri);
+        sendNotificationEmail(applicantNotificationRequest, uri);
     }
 
     private void sendNotificationEmail(NotificationRequest notificationRequest, URI uri) {
         HttpEntity<NotificationRequest> request = new HttpEntity<>(notificationRequest, buildHeaders());
         try {
             restTemplate.exchange(uri, HttpMethod.POST, request, String.class);
         } catch (Exception ex) {
-            log.error(MESSAGE,
-                notificationRequest.getCaseReferenceNumber(), MSG_SOLICITOR_EMAIL,
+            log.error(String.format("Failed to send email for case ID: %s for solicitor email: %s due to exception: %s",
+                notificationRequest.getCaseReferenceNumber(),
                 notificationRequest.getNotificationEmail(),
-                EXCEPTION, ex.getMessage());
+                ex.getMessage()));
         }
     }
 
-    private NotificationRequest createNotificationRequest(CallbackRequest callbackRequest) {
+    private NotificationRequest createNotificationRequestForAppSolicitor(CallbackRequest callbackRequest) {
         if (isConsentedApplication(callbackRequest.getCaseDetails())) {
-            notificationRequest = buildNotificationRequest(callbackRequest, SOLICITOR_REFERENCE,
+            applicantNotificationRequest = buildNotificationRequest(callbackRequest, SOLICITOR_REFERENCE,
                 CONSENTED_SOLICITOR_NAME, SOLICITOR_EMAIL, CONSENTED, GENERAL_EMAIL_BODY, DIVORCE_CASE_NUMBER);
         } else {
-            notificationRequest = buildNotificationRequest(callbackRequest, SOLICITOR_REFERENCE,
+            applicantNotificationRequest = buildNotificationRequest(callbackRequest, SOLICITOR_REFERENCE,
                 CONTESTED_SOLICITOR_NAME, CONTESTED_SOLICITOR_EMAIL, CONTESTED, GENERAL_EMAIL_BODY, DIVORCE_CASE_NUMBER);
         }
 
-        return notificationRequest;
+        return applicantNotificationRequest;
+    }
+
+    private NotificationRequest createNotificationRequestForRespSolicitor(CallbackRequest callbackRequest) {
```

Comment:
Currently unused - will be used in future tickets for resp sol notifications. Will be tested using unit tests generated as part of those future emails - exact same approach as currently taken by createNotificationRequestForAppSolicitor()

Reply:
can we cover unit tests?

Reply:
There's nothing using this method yet, so I had planned that this would be covered as part of that future logic. I can amend this new method and the existing one for the applicant and create unit tests then if you want

Reply:
okie, it is good to have at least one test for new method.

Reply:
Added new tests to cover this 👍